iam-login-service/src/main/java/it/infn/mw/iam/api/aup/model/AupRemindersAndSignatureValidator.java

```diff
@@ -50,55 +51,62 @@ public boolean isValid(AupDTO value, ConstraintValidatorContext context) {
       return false;
     }
 
-    if (aupRemindersInDays == null || aupRemindersInDays.isEmpty()) {
-      context.disableDefaultConstraintViolation();
-      context
-        .buildConstraintViolationWithTemplate(
-            "Invalid AUP: aupRemindersInDays cannot be empty or null")
-        .addConstraintViolation();
-      return false;
-    }
-
-    try {
-      List<Integer> numbers = Arrays.stream(aupRemindersInDays.split(","))
-        .map(String::trim)
-        .map(Integer::parseInt)
-        .collect(Collectors.toList());
-
-      if (numbers.stream().anyMatch(i -> i <= 0)) {
+    if (signatureValidityInDays == 0) {
+      if (aupRemindersInDays != null && !aupRemindersInDays.isEmpty()) {
         context.disableDefaultConstraintViolation();
         context
           .buildConstraintViolationWithTemplate(
-              "Invalid AUP: zero or negative values for reminders are not allowed")
+              "Invalid AUP: aupRemindersInDays cannot be set if signatureValidityInDays is 0")
           .addConstraintViolation();
         return false;
       }
+      return true;
+    }
 
-      if (numbers.stream().anyMatch(i -> i >= signatureValidityInDays)) {
-        context.disableDefaultConstraintViolation();
-        context
-          .buildConstraintViolationWithTemplate(
-              "Invalid AUP: aupRemindersInDays must be smaller than signatureValidityInDays")
-          .addConstraintViolation();
-        return false;
-      }
+    if (aupRemindersInDays != null) {
```

Comment:
at this point we're sure that signatureValidityInDays is greater than zero then we expect a value for aupRemindersInDays

I think we need something like if aupRemindersInDays is null add a constraint "Invalid AUP: aupRemindersInDays must be set when signatureValidityInDays is greater than zero"
We don't need to check empty value because probably with an empty string the code below will raise an exception